addons/product/models/product_template.py

```diff
@@ -381,9 +381,23 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
             if (not products) or (limit and (len(templates) > limit)):
                 break
 
+        searched_ids = set(templates.ids)
+        # some product.templates do not have product.products yet (dynamic variants configuration),
+        # we need to add the base _name_search to the results
+        # FIXME awa: this is really not performant at all but after discussing with the team
+        # we don't see another way to do it
+        if len(searched_ids) < limit:
+            searched_ids |= set([template_id[0] for template_id in
```

Comment:
You could have more results than the asked limit (if you had 79 on 80 then find 30 new ones).

Reply:
I still think that the 30 new ones should be added in the searched_ids and then re-sorted and re-limited by the last "_name_search".

Because if 10 of the 30 new ones actually come first when the last sorting is applied, they need to be in the searched ids for the result to be coherent.

Does that make sense?

Reply:
Indeed it makes sense. With that use case in mind, shouldn't we try to find limit results on templates with products (current behavior), then limit results on templates without products (your addition), then sort this 2xlimit and order it ? Unless we explicitly want templates without products only to fill the search if not results enough.

Reply:
In other words: remove the `if len(searched_ids) < limit:` ?

I think what makes the most sense is to have both templates with and without products sorted and limited in the results.
The "if" was there more for performance reasons but it could indeed give incoherent results so we should maybe remove it. 

This means we will always have 2 _name_search triggered to handle the "corner case" of dynamic products though so I'm a little conflicted...

What do you think?

Reply:
Well if it is a pure corner case I suppose avoiding an unnecessary query is a good idea. Ok then.